### ISSUE TITLE
Implement optional stacked multi-head attention

### ISSUE THREAD

**Issue by dingo-actual** [dingo-actual/om#15]
The current unstacked multi-head attention was implemented for memory-efficiency. However, this memory-efficiency only applies when gradients aren't being computed (i.e., a forward pass with `torch.no_grad()`). Stacking the attention matrices during training will improve CUDA utilization.
